test/testing/run.spec.ts

```diff
@@ -843,6 +849,106 @@ describe('Testing SDK', () => {
     });
   });
 
+  it('handles evaluators that implement both BaseTestEvaluator and BaseEventEvaluator', async () => {
+    type T = { x: number };
+    type O = string;
+
+    class MyEvaluator
+      extends BaseTestEvaluator<T, O>
+      implements BaseEventEvaluator
+    {
```

Comment:
i think what we can do here is provide a `BaseEvaluator` ABC that combines both so you don't need to do this "extends" and then "implements" workaround